### PR TITLE
JAMES-2186 GetMessages Manager only part

### PR DESCRIPTION
**Pull request by chibenwa**
Subset of https://github.com/linagora/james-project/pull/1046 for faster merge.

### CODE REVIEW

File: mailbox/api/src/test/java/org/apache/james/mailbox/fixture/MailboxFixture.java [hunk 1]
```diff
@@ -22,12 +22,12 @@
 import org.apache.james.mailbox.model.MailboxPath;
 
 public class MailboxFixture {
-    public static final String USER = "user";
-    public static final String OTHER_USER = "otheruser";
-    public static final String THIRD_USER = "thirdUser";
+    public static final String ALICE = "alice";
+    public static final String BOB = "bob";
+    public static final String CEDRIC = "cedric";
```
Comment (rouazana) [inline mailbox/api/src/test/java/org/apache/james/mailbox/fixture/MailboxFixture.java]: CAROL! :)

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java [hunk 2]
```diff
@@ -61,25 +62,76 @@
...
+        private final MessageMetaData messageMetaData;
+        private final MailboxId mailboxId;
+
+        public MetadataWithMailboxId(MessageMetaData messageMetaData, MailboxId mailboxId) {
+            this.messageMetaData = messageMetaData;
+            this.mailboxId = mailboxId;
+        }
+    }
+
+    private static class WrappedException extends RuntimeException {
```
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: we can replace this by Throwing functions

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java [hunk 3]
```diff
@@ -61,25 +62,76 @@
...
 import com.google.common.collect.Sets;
-import com.google.common.collect.Sets.SetView;
 
 public class StoreMessageIdManager implements MessageIdManager {
+
+    private static class MessageMoves {
+        private final ImmutableSet<MailboxId> previousMailboxIds;
+        private final ImmutableSet<MailboxId> targetMailboxIds;
+
+        MessageMoves(Collection<MailboxId> previousMailboxIds, Collection<MailboxId> targetMailboxIds) {
```
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: Can you give modifiers on this constructor and the methods, such as other embedded classes.

**Comment by rouazana** [comment]
test this please

File: mailbox/api/src/test/java/org/apache/james/mailbox/fixture/MailboxFixture.java (hunk 1 above)
Comment (chibenwa) [reply mailbox/api/src/test/java/org/apache/james/mailbox/fixture/MailboxFixture.java]: Whatever?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java (hunk 2 above)
Comment (chibenwa) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: It's just a class move, nothing to do with my code.

+ I don't see how Throwing functions coul help here....

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java (hunk 3 above)
Comment (chibenwa) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: :+1:

**Comment by chibenwa** [comment]
```
[aa5e00e8bf0f4280c00903755ceca5ad58401bf6] [ERROR] Errors: 
[aa5e00e8bf0f4280c00903755ceca5ad58401bf6] [ERROR]   ContainerTest.containerShouldBeReachableOnExposedPort:45 » Socket Connection r...
```

test this please

**Comment by chibenwa** [comment]
And again test this please on an other slave

**Comment by rouazana** [comment]
test this please

**Comment by rouazana** [comment]
test this please

**Comment by rouazana** [comment]
test this please

**Comment by rouazana** [comment]
[aa5e00e8bf0f4280c00903755ceca5ad58401bf6] [ERROR] Failures: 
[aa5e00e8bf0f4280c00903755ceca5ad58401bf6] [ERROR]   CassandraSetMessagesMethodTest>SetMessagesMethodTest.messageWithNonIndexableAttachmentShouldHaveItsEmailBodyIndexed:3811 1 expectation failed.
[aa5e00e8bf0f4280c00903755ceca5ad58401bf6] JSON path [0][1].messageIds doesn't match.
[aa5e00e8bf0f4280c00903755ceca5ad58401bf6] Expected: a collection containing "c5f9e760-b4e1-11e7-aeb1-1bac94e604ed"
[aa5e00e8bf0f4280c00903755ceca5ad58401bf6]   Actual: [c60e0ba0-b4e1-11e7-aeb1-1bac94e604ed]

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**